proxy/ProxySession.cc

```diff
@@ -227,6 +227,14 @@ ProxySession::set_inactivity_timeout(ink_hrtime timeout_in)
   }
 }
 
+void
+ProxySession::set_default_inactivity_timeout(ink_hrtime timeout_in)
+{
+  if (_vc) {
+    _vc->set_default_inactivity_timeout(timeout_in);
```

Comment:
to confirm, if _vc is of type UnixNetVConnection, would we not want it to update `next_inactivity_timeout_at` as that is no longer part of the function?

Reply:
This is a good question.

Note that before this patch, `set_default_inactivity_timeout` was called exactly once in UnixNet.cc. That was called off of NetEvent objects which were only ever `UnixNetVConnection` or `QUICNetVConnection` concrete types, both of which took the `UnixNetVConnection::set_default_inactivity_timeout` definition. In the previous use of this function, `set_default_inactivity_timeout` meant: (1) cancel any current activity_timeout_in, (2) set the next_inactivity_timeout_at to the provided default timeout value, and (3) set a boolean indicating that the current inactivity timeout is a default one. In other words, `set_default_inactivity_timeout` didn't mean "set the configured default inactivity timeout to this new value", it meant, "make the next timeout the provided value and indicate that it was set from the default inactivity timeout value". It was an ad-hoc implementation for the inactivity cop's single use of it.

This patch changes the meaning of the function to something more simple: "set the configured default_inactivity_timeout value to this new value". And now it is only called in:

1. In HttpTransact to provide a transaction level override.
2. In InactivityCop if no override has been set to set the global value.

So, to answer your question directly, we no longer set `next_inactivity_timeout_at` because the meaning of `set_default_inactivity_timeout` has changed to mean "set the configured default_inactivity_timeout", not the previous meaning of "configure an inactivity timeout value from this default value".

Reply:
I see, thanks for the explanation!

Reply:
Oh and I should add, in case someone reads this in the future, actually "setting" things to use the default inactivity timeout is done via the cancel_inactivity_timeout function. That cancels the explicit inactivity timeout, which effectively leads to falling back on the default inactivity timeout.